poetry/console/commands/shell.py

```diff
@@ -0,0 +1,52 @@
+from os import environ
+from distutils.util import strtobool
+
+from .venv_command import VenvCommand
+
+
+class ShellCommand(VenvCommand):
+    """
+    Spawns a shell within the virtual environment.
+
+    shell [options]
+    """
+
+    help = """The <info>shell</> command spawns a shell, according to the
+<comment>$SHELL</> environment variable, within the virtual environment.
+If one doesn't exist yet, it will be created.
+"""
+
+    def handle(self):
+        # Check if it's already activated or doesn't exist and won't be created
+        if strtobool(environ.get("POETRY_ACTIVE", "0")) or not self.venv.is_venv():
+            current_venv = environ.get("VIRTUAL_ENV")
+
+            if current_venv:
```

Comment:
Maybe we can do `if environ.get("VIRTUAL_ENV"):` here and drop the line 22?

Reply:
I use it again on line 27, though

Reply:
yeah, you're right... I should go to bed :)
**good job, man!**

PS Look into testing this code. I made a comment with some working example earlier today - check it out, and of course consult mighty [pytest docs](https://docs.pytest.org/en/latest/) or just ask questions if in trouble ;)

Reply:
Will do. Thanks for the support and good night ;)